### DISCUSSION TITLE
[Suggestion/Integration idea] "Or" chance based processing when cutting or loot table processing.

### DISCUSSION THREAD

**Discussion by Darkmega18** [vectorwing/FarmersDelight#755] [Ideas]
I'm currently fiddling with some recipes to do farmers delight styled breakdowns and stripping for bark for the other mods which add types of wood etc in my modpack in 1.19.2.
However I've come upon autumnity which has a log stripping logic where there seems to be around a third chance that the stripped log will become "sappy" rather than plain, the sap then being harvested by a bottle afterwards.

I could make a processing recipe, but it would mean it has to go one way or the other. Because I can't tell it to give me EITHER the sappy or nonsappy stripped log, only one or the other. and if I do a chance to do sappy and chance to do nonsappy together theres a chance I'll get none or both along with my bark. (also making the sap always pop out will make it somewhat op).

Is there a chance you can implement this kind of mechanic (or I just don't know it exists)? or even like, make something thats processed drop out a loot table or something? cause then I could just do up a loot table for it to spit out the log instead.

**Comment by thelegitdolt** [comment]
plz do this omg plz plz plz plz plz

**Comment by thelegitdolt** [comment]
plz

**Comment by Darkmega18** [comment]
Hahaha. thanks for your support, dolt. :)

**Comment by vectorwing** [comment]
@Darkmega18, moved the issue to Discussions, just letting you know. 👍

Exclusive outputs were considered at one point, yeah. In particular, it would've allowed me to make a more sensible gravel-to-flint recipe. Thing is, it was mostly archived due to a lack of time and patience, and also complexity to an extent. I'm still interested in trying it eventually, though.

Processing a loot table... could work, but might also be a huge point of complexity, given how diverse loot tables can be. Although I could just let the recipe call and roll a loot table .json, its conditions would only really accomodate the entity (user)'s state, which might make some things not work properly.

What I had originally considered was a toggle: a recipe would either roll outputs as AND (everything rolls with their own chances) or OR (only 1 output possible between all options, chance is weight). This has a different set of pros and cons.

**Comment by Darkmega18** [comment]
ah, alrighty. well one would probably do it mostly with small guaranteed loot tables, like even processing gravel into it's block break table, where it'll drop either gravel or flint per-se.
but if someone really wants to throw an entire chests worth of loot for smashing/chopping something, I mean sure they can. the complexity isn't maximum complex always, it's as complex as someone who wants to do shenanigans wants to make it.

so in my case it could be just that it's a table with 2 things in it: a stripped log or the sappy stripped log with lower weight + a table that always only gives 1 bark.

**Comment by Lethosos** [comment]
As someone who wrote his own data pack for FD's chopping board for different wood types (Chopper's Delight on my github,) I feel like, sometimes, you need to let that mod's mechanics do it for you. Yes, I would implement it if the mechanics are capable of it, but, as Vector points out, it's not possible to duplicate the chance of a sappy log on a cutting board.

That being said, just allowing multiple outputs with a simple chance modifier would simplify the JSON format. When I get a chance to, I'll post an example.

**Reply by Darkmega18** [reply]
if I knew choppers delight existed, I woulda just used yours! but too late now. I've done farmers style stripping and door/sign salvages, and also doing woodcutting recipes that take the stripped logs into more planks in a similar way to an older mod I remember that had a chopping block could could use to get more wood per log.

**Reply by Lethosos** [reply]
I strictly stick to vanilla and FD conventions for my pack, but I'm not concerned, it['s not that publicly advertised.

**Comment by Lethosos** [comment]
Using a random element from my data pack for `foobaz`'s sake:

`{
  "conditions": [
    {
      "type": "forge:mod_loaded",
      "modid": "byg"
    }
  ],
  "type": "farmersdelight:cutting",
  "ingredients": [
    {
      "item": "byg:aspen_log"
    }
  ],
  "tool": {
    "type": "farmersdelight:tool",
    "tool": "axe"
  },
  "result": [
    "random" [
       {
         "item": "byg:stripped_aspen_log",
         "weight": 3
       },
       {
         "item": "minecraft:diamond",
         "weight": 0.5
       }
    ],
    {
      "item": "farmersdelight:tree_bark"
    }
  ],
  "sound": "minecraft:item.axe.strip"
}`

This would mimic the weight system used in chest loot tables, for comparison.

**Reply by Darkmega18** [reply]
pain. compressed json. (yes I can uncompress it. but still)
~~btw, you missed a comma.~~ ;)
